source/grid/grid_generator.cc

```diff
@@ -3625,6 +3625,25 @@ namespace GridGenerator
     GridGenerator::merge_triangulations(
       tria_without_cylinder, cylinder_tria, tria, vertex_tolerance, true);
 
+    // Move the vertices in the middle of the faces of cylinder_tria slightly
+    // to give a better mesh quality. We choose to move them such that the
+    // distance between (0.05 + shell_region_width) and 0.1 is at least half
+    // as long as the remaining length to the boundary.
+    const double shifted_point =
+      std::min(0.05 + shell_region_width + 0.1 * std::sqrt(2) - 0.05 -
+                 shell_region_width,
+               0.1 * 4. / 3.);
+    for (const auto &cell : tria.active_cell_iterators())
+      for (unsigned int v = 0; v < GeometryInfo<2>::vertices_per_cell; ++v)
```

Comment:
It would be slightly simpler to just loop over the vertices directly, I think, e.g.,
```cpp
for (Point<2> &v : tria.get_vertices())
  if (v.distance()...)
```

Reply:
Nevermind: that is always a const reference.